### DISCUSSION TITLE
I don't see how to address this problem that pylance reports

### DISCUSSION THREAD

**Discussion by karthiknadig** [microsoft/pylance-release#2834] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/3840081/169194484-d4277919-41d1-436b-8afa-a1d4ce7cfb59.png)

![image](https://user-images.githubusercontent.com/3840081/169194579-417cc5f2-5eca-445e-9d36-8c2a04d5123e.png)

**Comment by erictraut** [comment, marked as answer]
Please provide a self-contained minimal code sample including import statements and the versions of the library or libraries that you are depending upon. It would also be useful if you could provide code in the form of text rather than screen shots.

Based on the symptoms here, it's likely that the base class (`WorkDoneProgressOptions`) contains one or more fields with default values. Dataclasses and dataclass-like libraries are not allowed to define fields without defaults after fields with defaults. That's because dataclass-like classes synthesize an `__init__` method with parameters that correspond to fields, and function signatures cannot contain parameters with default values before parameters without default values. For details refer to PEP 557, which says `TypeError will be raised if a field without a default value follows a field with a default value. This is true either when this occurs in a single class, or as a result of class inheritance`.

**Reply by karthiknadig** [reply]
Thanks that answers it. `@attrs.define` adds the synthesised `__init__` which was causing this.
